### DISCUSSION TITLE
Is ASYNCIFY_ONLY a way to go?

### DISCUSSION THREAD

**Discussion by iliakan** [emscripten-core/emscripten#18768] [Q&A]
Hello,

I tried ASYNCIFY, and the WASM binary grew +70%.

After enabling ASYNCIFY_ADVISE, it looks like almost everything is instrumented.

The most common reasons are:
- can change the state due to initial scan
- can change the state due to error
- can change the state due to die

I'm quite not getting it, could you please explain?

As far as I was reading, only the functions from the top of the stack till the actual async function needs instrumentation... Is that so?

In other words, can I analyze the calling sequence and list in ASYNCIFY_ONLY the functions that may be in the stack at the moment of the asyncified call?

E.g. `ASYNCIFY_ONLY=main,main_foo,main_foo_async`, assuming that `main_foo_async` can only be called from `main` or `main_foo` without other intermediate functions?

**Comment by iliakan** [comment]
The related discussion is at https://github.com/WebAssembly/binaryen/discussions/5495.

So far look like yes, only the functions from potential call stacks, leading to async execution, should be instrumented.

Figuring out how to avoid instrumenting too much ;)
